src/Profile.tsx

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import "./App.css";
+import { StyleService } from "./services/StyleService";
+
+import ServerService, {
+  LeaderBoardWindow,
+  Direction,
+} from "./services/ServerService";
+
+import { Button, NumericInput } from "@blueprintjs/core";
+import { FallingStarsView } from "./FallingStarsView";
+
+import "@blueprintjs/core/lib/css/blueprint.css";
+
+export class ProfilePage extends React.Component<{ user: string }, {}> {
+  user: any = null;
+  render() {
+    return (
+      <div className={stylesheet.outerDiv}>
+        <div className={stylesheet.topBar}>
+          <div className={stylesheet.profile}>
+            <div className={stylesheet.profileCircle}>
+              <h1>d</h1>
+            </div>
+          </div>
+        </div>
+
+        <div className="informationSection">
+          <h1>{this.props.user}</h1>
+        </div>
+        <div className="actionSection">
+          <Button
+            onClick={this.onClickMinusStars}
+            className={stylesheet.button}
+          >
+            Taketh
+          </Button>
+          <Button onClick={this.onClickAddStars} className={stylesheet.button}>
+            Giveth
+          </Button>
+        </div>
+        <pre style={{ overflow: "scroll" }}>
+          {JSON.stringify(this.user, null, 2)}
+        </pre>
+      </div>
+    );
+  }
+
+  componentDidMount = async () => {
```

Comment:
typically i use the functionName = () => {} notation to bind the function to the object. so that if its called from somewhere else, like as the callback in an onClick, "this" will behave as expected. I guess it doesnt hurt to do the componentDidMount also but I like being able to tell which functions on a class are likely callbacks based on if they are bound or not

Reply:
yee I see how that's helpful - I changed all of the lifecycle components into the `function ...` notion